common/dllserver/thorplugin.cpp

```diff
@@ -846,6 +846,29 @@ void SafePluginMap::loadFromList(const char * pluginsList)
     }
 }
 
+bool SafePluginMap::loadNamed(const char * pluginDirectories, const char * plugin)
+{
+    const char *pluginDir = pluginDirectories;
+    for (;*pluginDir;)
+    {
+        StringBuffer thisFile;
+        while (*pluginDir && *pluginDir != ENVSEPCHAR)
+            thisFile.append(*pluginDir++);
+        if(*pluginDir)
+            pluginDir++;
+
+        if(!thisFile.length())
+            continue;
+        Owned<IFile> file = createIFile(addPathSepChar(thisFile).append(plugin));
```

Comment:
The plugin "directory" list can also be a list of files.  Should this also check if endsWith(thisFile, plugin) - and then add in a similar way to loadFromList above

Reply:
Added